operator/src/main/java/org/bf2/operator/managers/MetricsManager.java

```diff
@@ -0,0 +1,121 @@
+package org.bf2.operator.managers;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.search.Search;
+import io.quarkus.runtime.Startup;
+import io.strimzi.api.kafka.KafkaList;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener;
+import org.bf2.common.ResourceInformer;
+import org.bf2.common.ResourceInformerFactory;
+import org.bf2.operator.operands.KafkaCluster;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import java.util.Locale;
+import java.util.stream.IntStream;
+
+import static org.bf2.operator.managers.InformerManager.filter;
+import static org.bf2.operator.operands.AbstractKafkaCluster.EXTERNAL_LISTENER_NAME;
+
+@Startup
+@ApplicationScoped
+public class MetricsManager {
+    static final String KAFKA_INSTANCE_BROKERS_DESIRED_COUNT = "kafka_instance_brokers_desired_count";
```

Comment:
We can have two different desired broker counts, this one and the one that we infer from the capabilities.  We may me need more precise naming here or in the other pr about what desired means.

Reply:
For clarity let me say, for this use-case, I need to publish the number of brokers that the kafka instance is supposed to have according to its configuration.  If an broker or an AZ happens to fail so that number of brokers is fewer, the user of the metric can tell this by comparing this desired value with the actual value reported by the kafka exporter.

Reply:
It's more about the terminology - we have two different desired replica counts.  Until we do quite a lot more work if this situation arises they will not be eventually consistent in any immediate sense.  I vote either for calling this one spec brokers, or changing the other other annotation to inferred or required brokers.